### PR TITLE
Allow preflight requests with OPTIONS method

### PR DESCRIPTION
**Pull request by luckysori**
This is useful because some browsers will send a preflight OPTIONS request as part of [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS), so that the server can respond whether it is acceptable to send the request with certain parameters.

Specifically, a preflight request to the comit node would use the `Access-Control-Request-Headers` and the `Access-Control-Request-Method` headers.

Additionally, related to #643, because it forces ledgers and assets to be serialized/deserialized using lowercase.

Requests to the comit node are only allowed from origin `localhost:3000` now. Issue #825 tracks allowing requests from `localhost` plus any port to be permitted.

### CODE REVIEW

File: application/comit_node/src/http_api/route_factory.rs [hunk 1]
```diff
@@ -73,14 +73,21 @@ pub fn create<T: MetadataStore<SwapId>, S: state_store::StateStore>(
         .and(warp::path::end())
         .and_then(http_api::routes::peers::get_peers);
 
-    rfc003_get_swap
+    warp::options()
```
Comment (thomaseizinger) [inline application/comit_node/src/http_api/route_factory.rs]: :x: This should be its own route similar to any others.

Comment (thomaseizinger) [reply application/comit_node/src/http_api/route_factory.rs]: By its own route I mean extract the two lines into a variable.

**Comment by luckysori** [comment]
> Thank you for this merge gift

It's not merged yet, you can merge yours first!!!!

**Comment by D4nte** [comment]
> > Thank you for this merge gift
> 
> It's not merged yet, you can merge yours first!!!!

Let's hope so then :)

**Comment by thomaseizinger** [comment]
I would hesitate to add a config field for now however, I would be okay with restricting the origin header to only allow `localhost` but with any port.

**Pull request merged**